app/models/attachment_data.rb

```diff
@@ -139,6 +139,15 @@ def draft_edition_for(user)
     draft_attachable.is_a?(Edition) ? draft_attachable : nil
   end
 
+  def draft_attachment
+    attachments.find { |attachment| attachment.attachable_type == "Edition" && attachment.attachable&.draft? }
```

Comment:
is the `&.` thing doing a check to make sure the attachable exists to stop it from falling over half way thorugh?

Reply:
Yeah, it is theoretically possible to have an orphaned attachment which isn't attached to anything (and presumably has been an issue in the past, as the other similar methods all include the safe navigation operator.